src/Education/XamlGenerator/DiffTranslator.cs

```diff
@@ -18,10 +18,81 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Text;
+using DiffPlex.DiffBuilder;
+using DiffPlex.DiffBuilder.Model;
+
 namespace SonarLint.VisualStudio.Education.XamlGenerator
 {
-    internal interface IDiffTranslator
+    public interface IDiffTranslator
     {
+        /// <summary>
+        /// Gets the difference between to strings in the format:
+        ///     Regular text one
+        ///     <Span Style="style">Text that is different</Span>
+        ///     Regular Text two
+        /// </summary>
         (string noncompliantXaml, string compliantXaml) GetDiffXaml(string noncompliantHtml, string compliantHtml);
     }
+
+    [Export(typeof(IDiffTranslator))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    internal class DiffTranslator : IDiffTranslator
+    {
+        private readonly IXamlWriterFactory xamlWriterFactory;
+
+        [ImportingConstructor]
+        public DiffTranslator(IXamlWriterFactory xamlWriterFactory)
```

Comment:
We could just make a new xamlWriter ourselves but I think it makes sense to just reuse this class